cmd/mayactl/app/command/volume_info.go

```diff
@@ -0,0 +1,102 @@
+package command
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	client "github.com/openebs/maya/pkg/client/jiva"
+	"github.com/openebs/maya/pkg/util"
+	"github.com/openebs/maya/types/v1"
+	"github.com/spf13/cobra"
+)
+
+var (
+	volumeInfoCommandHelpText = `
+	    Usage: mayactl volume info --volname <vol>
+
+            This command queries the info of the volume.
+        `
+)
+
+// CmdVolumeAccessOptions is used to store the value of flags used in the cli
+type CmdVolumeInfoOptions struct {
+	volName string
+}
+
+// NewCmdVolumeAccessMode shows info of OpenEBS Volume
+func NewCmdVolumeInfo() *cobra.Command {
+	options := CmdVolumeInfoOptions{}
+	cmd := &cobra.Command{
+		Use:     "info",
+		Short:   "Displays the info of Volume",
+		Long:    volumeInfoCommandHelpText,
+		Example: `mayactl volume info --volname vol`,
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(options.Validate(cmd), util.Fatal)
+			util.CheckErr(options.RunVolumeInfo(cmd), util.Fatal)
+		},
+	}
+	cmd.Flags().StringVarP(&options.volName, "volname", "n", options.volName,
+		"unique volume name.")
+	return cmd
+}
+
+// Validate verifies the command whether volName is passed or not.
+func (c *CmdVolumeInfoOptions) Validate(cmd *cobra.Command) error {
+	if c.volName == "" {
+		return errors.New("--volname is missing. Please specify a volume name created")
+	}
+	return nil
+}
+
+// TODO : Add more volume information
+// RunVolumeAccessMode runs info command and display the outputs in standard I/O.
+// Currently It displays volume access mode only.
+func (c *CmdVolumeInfoOptions) RunVolumeInfo(cmd *cobra.Command) error {
+	annotation := &Annotations{}
+	// GetVolumeAnnotation is called to get the volume controller's info such as
+	// controller's IP, status, iqn, replica IPs etc.
+	err := annotation.GetVolAnnotations(c.volName)
+	if err != nil {
+		fmt.Println("Can't get annotation, found error ", err)
+		return nil
+	}
+	if err != nil || annotation == nil {
+		fmt.Println(err)
+		return nil
+	}
+	if annotation.ControllerStatus != "Running" {
+		fmt.Println("Volume not reachable")
```

Comment:
The message hides important facts. It might help if controller status is available in the message.